### DISCUSSION TITLE
Hide an empty Column in Board View

### DISCUSSION THREAD

**Discussion by bronzeson** [community/community#10153] [Projects and Issues] [answered]
My columns are "backlog," "planned," "in progress," and "done"
When I'm in board view for my sprint, all the backlogs are already filtered out by nature of filtering for sprint.
Therefore, II'd like to hide the "Backlog" column in the board view.

**Comment by Harunbinissa** [comment]
Hi hello 👋

**Comment by MarkBennett** [comment]
Related, but not the same, it would be very cool if we could hide selected columns (like old iterations on the roadmap)

**Comment by rileybroughten** [comment, marked as answer]
Hello! 👋 You can hide a column on your board by clicking on the `...` menu, and again toggle which are visible from the `+` button to the right of your columns!

https://user-images.githubusercontent.com/101840513/225455659-7b71efa0-7348-436e-bb0e-61e887348818.mov

**Reply by ryders** [reply]
Hi @rileybroughten 

I appreciate that this is a workable solution, thank you. Seeing you might be the right person at GH to read this (Sr., huh? 😉), I would suggest this is a pretty bad UX design, especially when using columns based on time (columns are sprints).

### inconsistent behaviour
Generally, when grouping a table, any groups without any elements will not show. eg. grouping by assignee means all "possible assignees" to not show if they have nothing assigned. Ironically, grouping by Sprint also doesn't show sprints without any values in them.

### difficult to maintain
Furthermore, the idea of building a continuously changing, filter is sub-optimal. Starts fine with `-sprint:"Sprint 1"` but quickly goes down: my current board view has a filter like: `-sprint:"Sprint 218","Sprint 219","Sprint 220","Sprint 221","Sprint 222","Sprint 223","Sprint 224","Sprint 225","Sprint 226","Sprint 227","Sprint 226","Sprint 228","Sprint 229","Sprint 230","Sprint 231","Sprint 232","Sprint 233","Sprint 234"`. Note this essentially makes the search bar useless, so I can't quickly click and `assignee:@me` Plus the need to change the view with each sprint.

### possible filters
Much like the `sprint` filters (`@current`, `@next`, `@previous`) allow to be time-bound, I think a simple `sprint:@month` `sprint:@quarter` and `sprint:@year` would be a good starting point

Ideally, this really needs a `<attr>:@empty` and likewise a `-<attr>:@empty` filter, allowing to target empty or non-empty attributes? I did try `-sprint:""` and was left hoping! 🤣 I also realise this goes against another filter pattern for `no:assignee`. Maybe it's just a simpler `hide:empty` 🤷‍♀️ ok, I'll let you figure that one one!

my2c

**Reply by weitzhandler** [reply]
Hi @rileybroughten 
It's nice to have the ability to toggle visibility of individual columns, but it would be nice to enable auto-hiding empty columns. I've started [this discussion](https://github.com/orgs/community/discussions/118825).
